File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.py
class Solution:
    def nextGreatestLetter(self, ltrs: List[str], target: str) -> str:
        ans = 0
        lo, hi = 0, len(ltrs)-1
        while lo <= hi:
            mid = lo + ( hi-lo)//2
            if ltrs[mid]>target:
                ans = mid
                hi = mid-1
            else :
                lo = mid+1
        return ltrs[ans]

File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& ltrs, char target) {
        int ans =0;
        int lo = 0,hi= ltrs.size()-1;
        while(lo<=hi){
            int mid = lo +(hi-lo)/2;
            if(ltrs[mid]>target){
                ans = mid;
                hi = mid -1;
            }else{
                lo = mid +1;
            }
        }
        return ltrs[ans];
    }
};